resources/lib/library.py

```diff
@@ -1148,8 +1148,12 @@ def scriptPlaylists( self ):
 
                 if playlist.endswith( '-randomversion.xsp' ):
                     contents = xbmcvfs.File(playlistfile, 'r')
-                    contents_data = contents.read().decode('utf-8')
-                    xmldata = xmltree.fromstring(contents_data.encode('utf-8'))
+                    if sys.version_info.major == 3:
```

Comment:
(One for Mike - we've briefly discussed a Py3 specific version, as regards as Py2/3 compat version. Obv if you're still going for the former, this wouldn't be required...)